### ISSUE TITLE
Updating Contact Information failed (503 Invalid Parameter: X-NU-ACCEPT-REGISTRATION-TAC).

### ISSUE THREAD

**Issue by DennisSkov** [hexonet/whmcs-ispapi-registrar#218]
When we try updating the contact details for a client's .NU domain, we receive the mentioned error. 
These fields are available on the contact details page. 
![image](https://user-images.githubusercontent.com/5203348/142863682-3728b19a-06f7-41e8-838b-d68cd795de02.png)

Leaving them blank or filling them in correctly doesn't matter.

**Comment by DennisSkov** [comment]
This is the command (I've removed the query string from the bottom as it just contained all of this in a HTML-encoded string):
```
COMMAND = TradeDomain
DOMAIN = randolf.nu
OWNERCONTACT0FIRSTNAME = <REMOVED>
OWNERCONTACT0LASTNAME = <REMOVED>
OWNERCONTACT0ORGANIZATION = 
OWNERCONTACT0STREET = <REMOVED>
OWNERCONTACT0CITY = Lynge 
OWNERCONTACT0STATE = 
OWNERCONTACT0ZIP = 3540
OWNERCONTACT0COUNTRY = DK
OWNERCONTACT0PHONE = +45.<REMOVED>
OWNERCONTACT0FAX = 
OWNERCONTACT0EMAIL = <REMOVED>
ADMINCONTACT0FIRSTNAME = <REMOVED>
ADMINCONTACT0LASTNAME = <REMOVED>
ADMINCONTACT0ORGANIZATION = 
ADMINCONTACT0STREET = <REMOVED>
ADMINCONTACT0CITY = Lynge 
ADMINCONTACT0STATE = 
ADMINCONTACT0ZIP = 3540
ADMINCONTACT0COUNTRY = DK
ADMINCONTACT0PHONE = +45.<REMOVED>
ADMINCONTACT0FAX = 
ADMINCONTACT0EMAIL = <REMOVED>
TECHCONTACT0FIRSTNAME = <REMOVED>
TECHCONTACT0LASTNAME = <REMOVED>
TECHCONTACT0ORGANIZATION = 
TECHCONTACT0STREET = <REMOVED>
TECHCONTACT0CITY = Lynge 
TECHCONTACT0STATE = 
TECHCONTACT0ZIP = 3540
TECHCONTACT0COUNTRY = DK
TECHCONTACT0PHONE = +45.<REMOVED>
TECHCONTACT0FAX = 
TECHCONTACT0EMAIL = <REMOVED>
BILLINGCONTACT0FIRSTNAME = <REMOVED>
BILLINGCONTACT0LASTNAME = <REMOVED>
BILLINGCONTACT0ORGANIZATION = 
BILLINGCONTACT0STREET = <REMOVED>
BILLINGCONTACT0CITY = Lynge 
BILLINGCONTACT0STATE = 
BILLINGCONTACT0ZIP = 3540
BILLINGCONTACT0COUNTRY = DK
BILLINGCONTACT0PHONE = +45.<REMOVED>
BILLINGCONTACT0FAX = 
BILLINGCONTACT0EMAIL = <REMOVED>
X-NU-ACCEPT-REGISTRATION-TAC = 1
X-REGISTRANT-IDNUMBER = 260773XXXX
X-VATID = DK260773XXXX
```

The response is:
```
[RESPONSE]
code=503
description=Invalid Parameter: X-NU-ACCEPT-REGISTRATION-TAC
EOF
```

**Comment by papakai** [comment]
@DennisSkov it will take some days until I come to get this analyzed and patched.
In the meanwhile, please remove all these `X-*` parameters from command and execute it over our Frontend / Control Panel. See Widget "API Access" in your Dashboard there. This should be a working workaround until the patch is rolled out.

**Comment by DennisSkov** [comment]
Hi @papakai,

Thanks. We'll update the contact details through your control panel until the issue is fixed :-)

**Comment by papakai** [comment]
fyi, this is already in progress. expect a launch upcoming next week.

**Comment by papakai** [comment]
Hey @DennisSkov,

I have rolled out v7.2.9 which includes several further additional things like a generic caching mechanism for Zone Informations so that we don't have to again and again request QueryDomainOptions. For now with a TTL of 24h.
So, especially Registrar TLD Sync is a lot faster when accessing existing data cache (27s-35s to ~6-8s).

Check the upgrade instructions here:
https://github.com/hexonet/whmcs-ispapi-registrar/releases/tag/v7.2.9

I wasn't able to test if my bugfix for .nu really works, because .nu is not working in our OT&E. Still, my expectation is -> patched!

HTH
Kai

**Comment by DennisSkov** [comment]
Hi @papakai 

I just updated - unfortunately it didn't solve our issue.
When updating, we still get the following reply on TradeDomain:
```
[RESPONSE]
code=503
description=Invalid Parameter: X-NU-ACCEPT-REGISTRATION-TAC
EOF

```

**Comment by papakai** [comment]
ok, thanks @DennisSkov. I'll check that again immediately tomorrow.

**Comment by papakai** [comment]
fixed in v7.2.10.

Have a nice day!

**Comment by DennisSkov** [comment]
Almost! Another error now :-D

```
[RESPONSE]
DESCRIPTION=Invalid attribute value; INVALID CONTACT [OWNERCONTACT (Missing required attribute; X-REGISTRANT-IDNUMBER)]
QUEUETIME=0
CODE=541
RUNTIME=0.18

QUEUETIME=0
RUNTIME=0.266

EOF
```
X-REGISTRANT-IDNUMBER is set on the domain already. The TradeDomain command doesn't send X-REGISTRANT-IDNUMBER, though.

**Comment by papakai** [comment]
Great! Looks like the .NU documentation in our wiki is incomplete... Something I couldn't test further (I used a live domain from enavn to see my domain fields removal works and it did, but of course I blocked the submission of the command to the API to not get your domain accidentally updated). Such things are actually not testable on my end regarding .nu.
Anyways, step-by-step! Thanks for forwarding.
